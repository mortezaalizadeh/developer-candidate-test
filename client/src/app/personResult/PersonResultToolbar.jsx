import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import DeleteIcon from '@material-ui/icons/Delete';
import Styles from './Styles';

const PersonResultToolbar = ({ classes, numSelected, onDeleteButtonClicked }) => (
  <Toolbar
    className={classNames(classes.toolbarRoot, {
      [classes.toolbarHighlight]: numSelected > 0,
    })}>
    <div className={classes.toolbarTitle}>
      {numSelected > 0 ? (
        <Typography color="inherit" variant="subtitle1">
          {numSelected + ' selected'}
        </Typography>
      ) : (
        <Typography variant="h6" id="tableTitle">
          Select person(s) to remove
        </Typography>
      )}
    </div>
    <div className={classes.toolbarSpacer} />
    <div className={classes.toolbarActions}>
      {numSelected > 0 && (
        <Tooltip title="Delete">
          <IconButton aria-label="Delete" onClick={onDeleteButtonClicked}>
            <DeleteIcon />
          </IconButton>
        </Tooltip>
      )}
    </div>
  </Toolbar>
);

PersonResultToolbar.propTypes = {
  classes: PropTypes.object.isRequired,
  numSelected: PropTypes.number.isRequired,
  onDeleteButtonClicked: PropTypes.func.isRequired,
};

export default withStyles(Styles)(PersonResultToolbar);
