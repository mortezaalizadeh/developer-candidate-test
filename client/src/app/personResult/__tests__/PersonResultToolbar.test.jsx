import React from 'react';
import renderer from 'react-test-renderer';
import PersonResultToolbar from '../PersonResultToolbar';

it('should render correctly', () => {
  const tree = renderer.create(<PersonResultToolbar numSelected={10} />).toJSON();

  expect(tree).toMatchSnapshot();
});
