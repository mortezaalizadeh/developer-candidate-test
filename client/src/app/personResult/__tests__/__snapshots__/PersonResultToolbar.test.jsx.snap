// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="MuiToolbar-root-9 MuiToolbar-regular-11 MuiToolbar-gutters-10 Component-toolbarRoot-4 Component-toolbarHighlight-5"
>
  <div
    className="Component-toolbarTitle-8"
  >
    <h6
      className="MuiTypography-root-13 MuiTypography-subtitle1-31 MuiTypography-colorInherit-42"
    >
      10 selected.label
    </h6>
  </div>
  <div
    className="Component-toolbarSpacer-6"
  />
  <div
    className="Component-toolbarActions-7"
  >
    <button
      aria-describedby={null}
      aria-label="Delete"
      className="MuiButtonBase-root-63 MuiIconButton-root-57"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      title="delete.button.caption"
      type="button"
    >
      <span
        className="MuiIconButton-label-62"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-66"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-75"
      />
    </button>
  </div>
</div>
`;
